src/main.rs

```diff
@@ -43,30 +48,31 @@ fn print_connector_info(gpu: &Card, resources: &ResourceHandles) {
             .get_connector(*connector_handle, force_probe)
             .expect("Failed to get GPU connector info");
 
-        let connector_interface = connector.interface().as_str();
-        let interface_id = connector.interface_id();
+        let name = get_connector_name(&connector);
 
         let connection_state = match connector.state() {
             ConnectorState::Connected => "✅",
             ConnectorState::Disconnected => "❌",
             ConnectorState::Unknown => "❔",
         };
 
-        println!("\t{connector_interface}-{interface_id}, Connected={connection_state}");
+        println!("\t{name}, Connected={connection_state}");
     }
 }
 
-fn first_connected_connector(gpu: &Card) -> Option<ConnectorHandle> {
+fn get_connected_connector(gpu: &Card, name: &Option<String>) -> Option<ConnectorHandle> {
     gpu.resource_handles()
         .expect("Failed to get GPU resource handles")
         .connectors
         .iter()
         .find(|connector_handle| {
             let force_probe = false;
-            gpu.get_connector(**connector_handle, force_probe)
-                .expect("Failed to get GPU connector info")
-                .state()
-                == ConnectorState::Connected
+            let info = gpu
+                .get_connector(**connector_handle, force_probe)
+                .expect("Failed to get GPU connector info");
+            let name_matches =
+                if let Some(name) = name { name == &get_connector_name(&info) } else { true };
+            name_matches && info.state() == ConnectorState::Connected
```

Comment:
Just for fun, throwing my suggestion into the pile :)

```rust
gpu.resource_handles()
    .expect("Failed to get GPU resource handles")
    .connectors
    .iter()
    .map(|connector_handle| {
        let force_probe = false;
        gpu.get_connector(*connector_handle, force_probe).expect("Failed to get GPU connector info")
    })
    .filter(|info| {
        info.state() == ConnectorState::Connected
    })
    .find(|info| {
        if let Some(name) = name {
            name == &get_connector_name(info)
        } else {
            true
        }
    })
    .map(|info| {
        info.handle()
    })
```

Reply:
Neat :+1: . I ended up rewriting the whole thing to return an iterator over connected connectors and separated further name filtering to main. I like it better now but I still don't love it, all of it feels very ad-hoc, not like a proper library code. Speaking of which, it would be nice to start a library crate once we settle on better abstractions :smile:

Reply:
@bschwind nice, cool it is possible to write this in one "flow". It may be less elegant to give user proper error (based on "no connected output" vs. "the output you've specified does not exist or is not connected").

As a form of exercise, I've posted a suggestion on top of @mbernat's latest code.